eureka-client/src/main/java/com/netflix/discovery/util/DeserializerStringCache.java

```diff
@@ -0,0 +1,77 @@
+package com.netflix.discovery.util;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.cfg.ContextAttributes;
+
+/**
+ * A non-locking alternative to {@link String#intern()} and {@link StringCache} that works with DeserializationContext.  Definitely
+ * NOT thread-safe.
```

Comment:
Should ConcurrentHashMap be used instead? or if thread safety doesn't matter since this is a cache and misses here aren't a big deal update the docs to call that out. As it is written it implies that the lack safety could be an issue.